zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/FileSystemStorage.java

```diff
@@ -60,16 +59,25 @@
   public FileSystemStorage(ZeppelinConfiguration zConf, String path) throws IOException {
     this.zConf = zConf;
     this.hadoopConf = new Configuration();
+    URI zepConfigURI;
+
+    zepConfigURI = getURI(path, "Failed to get URI path");
     // disable checksum for local file system. because interpreter.json may be updated by
     // non-hadoop filesystem api
     // disable caching for file:// scheme to avoid getting LocalFS which does CRC checks
+
     this.hadoopConf.setBoolean("fs.file.impl.disable.cache", true);
-    this.hadoopConf.set("fs.file.impl", RawLocalFileSystem.class.getName());
```

Comment:
Don't remove this line, see the comment above.

Reply:
We have set **zeppelin.notebook.dir** and **zeppelin.config.fs.dir** to s3a FileSystemStorage.

If we don't remove this line then it throws exception
``` WARN [2019-07-29 22:47:49,335] ({main} ContextHandler.java[log]:2359) - unavailable
MultiException stack 1 of 1
java.lang.ClassCastException: org.apache.hadoop.fs.RawLocalFileSystem cannot be cast to org.apache.hadoop.fs.LocalFileSystem
        at org.apache.hadoop.fs.FileSystem.getLocal(FileSystem.java:447)
        at org.apache.hadoop.fs.LocalDirAllocator$AllocatorPerContext.confChanged(LocalDirAllocator.java:301)
        at org.apache.hadoop.fs.LocalDirAllocator$AllocatorPerContext.getLocalPathForWrite(LocalDirAllocator.java:378)
        at org.apache.hadoop.fs.LocalDirAllocator$AllocatorPerContext.createTmpFileForWrite(LocalDirAllocator.java:461)
        at org.apache.hadoop.fs.LocalDirAllocator.createTmpFileForWrite(LocalDirAllocator.java:200)
        at org.apache.hadoop.fs.s3a.S3AFileSystem.createTmpFileForWrite(S3AFileSystem.java:753)
        at org.apache.hadoop.fs.s3a.S3ADataBlocks$DiskBlockFactory.create(S3ADataBlocks.java:811)
        at org.apache.hadoop.fs.s3a.S3ABlockOutputStream.createBlockIfNeeded(S3ABlockOutputStream.java:190)
        at org.apache.hadoop.fs.s3a.S3ABlockOutputStream.<init>(S3ABlockOutputStream.java:168)
        at org.apache.hadoop.fs.s3a.S3AFileSystem.create(S3AFileSystem.java:1044)
        at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:1130)
        at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:1110)
        at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:999)
        at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:987)
```

Reply:
I am afraid this approach will introduce another bug which is mentioned in the above PR. how about setting it in your `hdfs-site.xml`?

Reply:
Sorry I did not get which bug you are mentioning?

Reply:
See this comment https://github.com/apache/zeppelin/pull/3411/files#diff-6b930e3e153d78dacf931917ff89587cR65

Reply:
> how about setting it in your hdfs-site.xml?
Which property we have to set?

Reply:
set `fs.file.impl` to `org.apache.hadoop.fs.LocalFileSystem`

Reply:
After setting the above property still it throws same exception.

Reply:
hmm, right, it will be overwritten here. Let me think about it

Reply:
@zjffdu Did you get a chance to look at it this issue

Reply:
@bhavikpatel9977 One approach I can think of is adding one extra configuration such as `zeppelin.config.storage.useS3a`, by default it is false. 
```
if (!useS3a) {
  this.hadoopConf.set("fs.file.impl", RawLocalFileSystem.class.getName());
}
```

This way we can not only solve your issue but also won't break other user cases.

Reply:
Ok. We are addding `zeppelin.config.storage.useS3A` property in conf/zeppelin-site.xml.template with default value as false.

Reply:
Rather then adding new extra property we can also do something like this 
```if (!isS3AFileSystem(defaultFSURI, zepConfigURI)) {
  this.hadoopConf.set("fs.file.impl", RawLocalFileSystem.class.getName());
}```

``` public boolean isS3AFileSystem(URI defaultFSURI, URI zepConfigURI) {
    return defaultFSURI.getScheme().equals(S3A)
        || (StringUtils.isNotEmpty(zepConfigURI.getScheme())
        && zepConfigURI.getScheme().equals(S3A));
  }```

Reply:
hmm, this approach seems better, as it doesn't introduce new configuration. BTW, please add a comment to explain this change.